stockroom/storages/generic.py

```diff
@@ -0,0 +1,30 @@
+from .. import parser
+from ..repository import StockRepository
+
+
+# TODO: Maybe need another name than `GenericStore` but keeping it-
+# since scope of the name MetricStore seems to be too narrow
+
+class GenericStore:
+    """
+    A generic data storage class that does the functionality of
+    hangar metadata
+    """
```

Comment:
This isn't really a good name for this at all. Even though it's explained in the docstring, it relies on the user knowing that hangar metadata is a key/value store for text. Maybe something like `InfoStore` would be better? But even that could be confusing to use. 

I'm thinking what needs to happen here is that this `FooStore` class needs to implement a few checks inside the `save` method. If the `value` is some `str` type, then it is saved as normal, if it is `int`, then the store adds some character to identify it as an integer value so that it can be converted to the appropriate type upon reading it back. Same thing for `float` and `bytes` type data. If it's anything else, then it should raise some kind of `ValueError` indicating that it is not a valid object for this type of store.

Reply:
That seems to be a reasonable thing to do. Will add to the TODO of next minor. I still need to find a better name.